packages/cart/src/models/line-item.ts

```diff
@@ -23,23 +29,22 @@ type OptionalLineItemProps =
   | "compare_at_unit_price"
   | "requires_shipping"
   | "cart"
-  | DAL.EntityDateColumns
+  | DAL.SoftDeletableEntityDateColumns
 
 @Entity({ tableName: "cart_line_item" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
 export default class LineItem {
   [OptionalProps]?: OptionalLineItemProps
 
   @PrimaryKey({ columnType: "text" })
   id: string
 
-  @Property({ columnType: "text" })
+  @Property({ columnType: "text", index: "IDX_line_item_cart_id" })
```

Comment:
todo: since the relation is persist false, it will remove the foreign key constraint from the db, maybe that we don't really care but otherwise you need to apply the many to one on the foregin key prop as well, we can discuss if needed

Reply:
We prob. don't want this persist flag here actually. Will revert.